pycalphad/core/minimizer.pyx

```diff
@@ -96,53 +110,55 @@ cdef void write_row_fixed_mole_fraction(double[:] out_row, double* out_rhs, int
                                         double[:, ::1] mass_jac, double[:, ::1] c_component,
                                         double[:, ::1] c_statevars, double[::1] c_G, double[:, ::1] masses,
                                         double moles_normalization, double[::1] moles_normalization_grad,
-                                        double[::1] phase_amt, int idx):
+                                        double[::1] phase_amt, int idx, double prefactor):
+    if prefactor == 0.0:
+        return
```

Comment:
Implicitly requires that `out_row` is initialized to zeros. This is currently true, but if we change the behavior, e.g. `equilibrium_matrix = np.empty` instead of `np.zeros`, breaks. 

having the shortcut here is nice, but maybe we just loop over outrow and set to zero explicitly rather than return

Reply:
For linear combination conditions, `write_row_fixed_mole_fraction` is called multiple times in a loop on the same memory. The caller is responsible for initializing whatever `out_row` points to.